### PR TITLE
Revamp Azure AD connector with more functions 

### PR DESCRIPTION
**Pull request by sachinira**
Resolves https://github.com/wso2-enterprise/choreo/issues/6311
Resolves https://github.com/wso2-enterprise/choreo/issues/6388
fixes https://github.com/wso2-enterprise/choreo/issues/6657

## Automation tests
 - Unit tests 
   > Tested
 - Integration tests
   > Tested

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

 
https://docs.microsoft.com/en-us/graph/api/resources/teams-api-overview?view=graph-rest-1.0
https://docs.microsoft.com/en-us/graph/azuread-identity-access-management-concept-overview
Azure AD connector in Logic Apps
https://docs.microsoft.com/en-us/connectors/azuread/

### CODE REVIEW

File: aad/Dependencies.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+[[dependency]]
+org = "ballerina"
+name = "url"
+version = "1.1.0-beta.1"
+
+[[dependency]]
+org = "ballerina"
+name = "mime"
+version = "1.1.0-beta.1"
```
Comment (LakshanSS) [inline aad/Dependencies.toml:49]: Since the latest version is beta2, please change everything to `beta.2`

File: aad/client.bal [hunk 2]
```diff
@@ -0,0 +1,423 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (LakshanSS) [inline aad/client.bal:1]: If we are adding completely new file, we can change the year as 2021

File: aad/Dependencies.toml (hunk 1 above)
Comment (sachinira) [reply aad/Dependencies.toml]: Updated to beta 2. Won't we release the connector in beta 1?

File: aad/client.bal (hunk 2 above)
Comment (sachinira) [reply aad/client.bal]: This was a new file. So changed the year thanks

File: aad/Dependencies.toml (hunk 1 above)
Comment (LakshanSS) [reply aad/Dependencies.toml]: Nope. No release in beta1. We might release the connector in beta2 or sometimes we'll go to beta3 directly. ;)

File: aad/Ballerina.toml [hunk 3]
```diff
@@ -1,7 +1,11 @@
-[project]
-org-name= "ballerinax"
-version= "0.9.0"
-license= ["Apache-2.0"]
+[package]
+org= "ballerinax"
+name= "azure.ad"
+version= "1.0.0"
```
Comment (LakshanSS) [inline aad/Ballerina.toml:4]: Connector version should be in SNAPSHOT in the master branch

File: aad/client.bal [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+    # List information about users.
+    # 
+    # + queryParams - Optional query parameters. This method support OData query parameters to customize the response.
+    #                 It should be an array of type `string` in the format `<QUERY_PARAMETER_NAME>=<PARAMETER_VALUE>`
+    #                 **Note:** For more information about query parameters, refer here: 
+    #                   https://docs.microsoft.com/en-us/graph/query-parameters
+    # + return - A stream of type `User` if sucess. Else `error`.
+    @display {label: "List Users"}
+    remote isolated function listUsers(@display {label: "Optional Query Parameters"} string[] queryParams = []) 
+                                       returns stream<User, error?>|error {
```
Comment (Chuhaa) [inline aad/client.bal:93]: Can you add  display annotation for stream?

File: aad/client.bal [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+    # + objectId - The object ID
+    # + queryParams - Optional query parameters. This method support OData query parameters to customize the response.
+    #                 It should be an array of type `string` in the format `<QUERY_PARAMETER_NAME>=<PARAMETER_VALUE>`
+    #                 **Note:** For more information about query parameters, refer here: 
+    #                   https://docs.microsoft.com/en-us/graph/query-parameters
+    # + return - A stream of type `Group` if sucess. Else `error`.
+    @display {label: "List Parent Groups"}
+    remote isolated function listParentGroups(@display {label: "Directory Object Type"} DirctoryObject 'type, 
+                                              @display {label: "Object ID"} string objectId, 
+                                              @display {label: "Optional Query Parameters"} string[] queryParams = [])
```
Comment (Chuhaa) [inline aad/client.bal:216]: Shall we get query params as string?

File: aad/client.bal [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+    #                   https://docs.microsoft.com/en-us/graph/query-parameters
+    # + return - A stream of type `Group` if sucess. Else `error`.
+    @display {label: "List Transitive Parent Groups"}
+    remote isolated function listTransitiveParentGroups(@display {label: "Directory Object Type"} DirctoryObject 'type, 
+                                                        @display {label: "Object ID"} string objectId, 
+                                                        @display {label: "Optional Query Parameters"} 
+                                                        string[] queryParams = []) returns 
+                                                        stream<Group, error?>|error {
+        string path = EMPTY_STRING;
+        if('type == "user") {
```
Comment (Chuhaa) [inline aad/client.bal:249]: In some places USER is used. Shall we use that here?

File: aad/Ballerina.toml (hunk 3 above)
Comment (sachinira) [reply aad/Ballerina.toml]: Fixed

File: aad/Module.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+easily. It provides capability to perform management operations on `User` and `Group` resources in an Azure AD tenent 
+(Organization).
+
+The connector is developed on top of Microsoft Graph is a REST web API that empowers you to access Microsoft Cloud 
+service resources. This version of the connector only supports the operations on users and groups of an Azure AD
+ 
+This module supports **Ballerina SL Beta 2**  version.
+ 
+## Obtaining tokens
+Follow the following steps below to obtain the configurations.
```
[the remark this answers is not in the document]
Comment (sachinira) [reply aad/Module.md]: Added

File: aad/client.bal [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+# 
+# + httpClient - the HTTP Client
+@display {
+    label: "Azure AD Client", 
+    iconPath: "MSAzureADLogo.svg"
+}
+public client class Client {
+    http:Client httpClient;
+
+    public isolated function init(Configuration config) returns error? {
```
[the remark this answers is not in the document]
Comment (sachinira) [reply aad/client.bal]: Added

File: aad/client.bal (hunk 5 above)
Comment (sachinira) [reply aad/client.bal]: Fixed this

File: aad/client.bal (hunk 6 above)
Comment (sachinira) [reply aad/client.bal]: Fixed thanks

File: aad/client.bal (hunk 4 above)
Comment (sachinira) [reply aad/client.bal]: Added them. Thanks.

File: aad/Package.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+#### Compatibility
+|                               | Version                       |
+|-------------------------------|-------------------------------|
+| Ballerina Language Version    | **Swan Lake Beta2**           |
+| API Version                   | **Microsoft Graph API v1.0**  |
+
+
+### Report Issues
+To report bugs, request new features, start new discussions, view project boards, etc., go to the [https://github.com/ballerina-platform/module-ballerinax-azure.ad](link)
```
Comment (Chuhaa) [inline aad/Package.md:15]:
```suggestion
To report bugs, request new features, start new discussions, view project boards, etc., go to the [Ballerina Microsoft Azure AD repository](https://github.com/ballerina-platform/module-ballerinax-azure.ad)
```

**Review by abeykoon — APPROVED** [review]

File: aad/Package.md (hunk 9 above)
Comment (sachinira) [reply aad/Package.md]: Fixed this. thanks

**Review by abeykoon — APPROVED** [review]

**Pull request merged**